hyper/http20/stream.py

```diff
@@ -268,8 +272,11 @@ def open(self, end):
         The `end` flag controls whether this will be the end of the stream, or
         whether data will follow.
         """
+        headers = list(self.headers.items())
```

Comment:
There should be no need for this conversion: the HTTPHeaderMap iterates like a list of two-tuples.